registration/include/pcl/registration/impl/transformation_estimation_symmetric_point_to_plane_lls.hpp

```diff
@@ -0,0 +1,251 @@
+  transformation_matrix (0, 2) = static_cast<Scalar> ( sin (gamma) * sin (alpha) + std::cos (gamma) * sin (beta) * std::cos (alpha));
+  transformation_matrix (1, 0) = static_cast<Scalar> ( sin (gamma) * std::cos (beta));
+  transformation_matrix (1, 1) = static_cast<Scalar> ( std::cos (gamma) * std::cos (alpha) + sin (gamma) * sin (beta) * sin (alpha));
+  transformation_matrix (1, 2) = static_cast<Scalar> (-std::cos (gamma) * sin (alpha) + sin (gamma) * sin (beta) * std::cos (alpha));
+  transformation_matrix (2, 0) = static_cast<Scalar> (-sin (beta));
+  transformation_matrix (2, 1) = static_cast<Scalar> ( std::cos (beta) * sin (alpha));
+  transformation_matrix (2, 2) = static_cast<Scalar> ( std::cos (beta) * std::cos (alpha));
+
+  transformation_matrix (0, 3) = static_cast<Scalar> (tx);
+  transformation_matrix (1, 3) = static_cast<Scalar> (ty);
+  transformation_matrix (2, 3) = static_cast<Scalar> (tz);
+  transformation_matrix (3, 3) = static_cast<Scalar> (1);
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////
+template <typename PointSource, typename PointTarget, typename Scalar> inline void
+pcl::registration::TransformationEstimationSymmetricPointToPlaneLLS<PointSource, PointTarget, Scalar>::
+estimateRigidTransformation (ConstCloudIterator<PointSource>& source_it, ConstCloudIterator<PointTarget>& target_it, Matrix4 &transformation_matrix) const
+{
+  using Vector6d = Eigen::Matrix<double, 6, 1>;
+  using Matrix6d = Eigen::Matrix<double, 6, 6>;
+
+  Matrix6d ATA;
+  Vector6d ATb;
+  ATA.setZero ();
+  ATb.setZero ();
+
+  // Approximate as a linear least squares problem
+  while (source_it.isValid () && target_it.isValid ())
+  {
+    if (!std::isfinite (source_it->x) ||
+        !std::isfinite (source_it->y) ||
+        !std::isfinite (source_it->z) ||
+        !std::isfinite (target_it->x) ||
+        !std::isfinite (target_it->y) ||
+        !std::isfinite (target_it->z) ||
+        !std::isfinite (source_it->normal_x) ||
+        !std::isfinite (source_it->normal_y) ||
+        !std::isfinite (source_it->normal_z) ||
+        !std::isfinite (target_it->normal_x) ||
+        !std::isfinite (target_it->normal_y) ||
+        !std::isfinite (target_it->normal_z))
+    {
+      ++target_it;
+      ++source_it;    
+      continue;
+    }
+
+    const float & sx = source_it->x;
+    const float & sy = source_it->y;
+    const float & sz = source_it->z;
+    const float & dx = target_it->x;
+    const float & dy = target_it->y;
+    const float & dz = target_it->z;
+    const float & nx = source_it->normal[0] + target_it->normal[0];
+    const float & ny = source_it->normal[1] + target_it->normal[1];
+    const float & nz = source_it->normal[2] + target_it->normal[2];
+
+    double a = nz*sy - ny*sz;
+    double b = nx*sz - nz*sx; 
+    double c = ny*sx - nx*sy;
+   
+    //    0  1  2  3  4  5
```

Comment:
Why not simply use `TriangularView` to show intent and then `(x,y)` to refer to rows and columns?

Reply:
Good idea, changed.